src/transformers/modeling_utils.py

```diff
@@ -4805,13 +4806,29 @@ def forward(
         return output
 
 
-def unwrap_model(model: nn.Module) -> nn.Module:
+def unwrap_model(model: nn.Module, recursive: bool = False) -> nn.Module:
     """
     Recursively unwraps a model from potential containers (as used in distributed training).
 
     Args:
         model (`torch.nn.Module`): The model to unwrap.
+        recursive (`bool`, *optional*, defaults to `False`):
+            Whether to recursively extract all cases of `module.module` from `model` as well as unwrap child sublayers
+            recursively, not just the top-level distributed containers.
     """
+    # Use accelerate implementation if available (should always be the case when using torch)
+    # This is for pytorch, as we also have to handle things like dynamo
+    if is_accelerate_available():
+        kwargs = {}
+        if recursive:
+            if not is_accelerate_available("0.29.0"):
+                raise RuntimeError(
+                    "Setting `recursive=True` to `unwrap_model` requires `accelerate` v0.29.0. Please upgrade your version of accelerate"
+                )
+            else:
+                kwargs["recursive"] = recursive
+        return extract_model_from_parallel(model, **kwargs)
```

Comment:
nit nit - it's nice to do an early return and not have the main logic indented 

```suggestion
    # Use accelerate implementation if available (should always be the case when using torch)
    # This is for pytorch, as we also have to handle things like dynamo
    if not is_accelerate_available():
        return
        
    kwargs = {}
    if recursive:
        if not is_accelerate_available("0.29.0"):
            raise RuntimeError(
                "Setting `recursive=True` to `unwrap_model` requires `accelerate` v0.29.0. Please upgrade your version of accelerate"
            )
        else:
            kwargs["recursive"] = recursive
    return extract_model_from_parallel(model, **kwargs)
```

Reply:
We can't return early, as there's logic involved for if we're not using accelerate :) (see just below it)

Reply:
(Though @SunMarc we can probably refactor and move the not version up first)

Reply:
Ah, sorry, my bad!